backend/dinesmart/views.py

```diff
@@ -0,0 +1,171 @@
+from django.shortcuts import render
+from django.http import HttpResponse, JsonResponse
+from secrets import token_bytes
+import hmac
+import hashlib
+import string
+import time
+import json
+from django.views.decorators.csrf import csrf_exempt
+
+from dinesmart.models import Users, UserAuthTokens, PasswordReset
+
+# Create your views here.
+
+@csrf_exempt
+def create_user(request):
+    #only accept post requests
+    if request.method != "POST":
+        return HttpResponse("only POST calls accepted", status=404)
+
+    #input validation
+    try:
+        payload = json.loads(request.body)
+        email = payload["email"]
+        password = payload["password"]
+    except:
+        return HttpResponse("missing/blank email or password", status=401)
+
+    if (email == "" or password == ""):
+        return HttpResponse("missing/blank email or password", status=401)
+
+    # make sure email isn't already taken
+    try:
+        user_count = Users.objects.filter(email=email).count() 
+        if user_count != 0:
+            return HttpResponse("email already in use", status=406)
+    except Exception as e:
+        return HttpResponse(e, status=401)
+    
+    # save user in database
+    user = Users(email=email, password=password)
+    try:
+        user.save()
+        #create authentication token for session
+        token = createAuthToken(email)
+
+        request.session['email'] = email
+        request.session['token'] = str(token)
+        request.session.set_expiry(1800) #session expires in 1800 seconds = 0.5 hour
+        return HttpResponse("user succesfully created", status=201)
+    except:
+        return HttpResponse("error saving user", status=401)
+
+@csrf_exempt
+def login(request):
+    # only accept post requests
+    if request.method != "POST":
+        return HttpResponse("only POST calls accepted", status=404)
+
+    # input validation
+    try:
+        payload = json.loads(request.body)
+        email = payload['email']
+        password = payload['password']
+    except:
+        return HttpResponse("missing/blank email or password", status=401)
+
+    # make sure user exists
+    try:
+        user_count = Users.objects.filter(email=email).count()
+        if user_count != 1:
+            return HttpResponse("unable to find user", status=404)
+    except Exception as e:
+        return HttpResponse(e, status=404)
+
+    #create authentication token for session
+    token = createAuthToken(email)
+
+    request.session['email'] = email
+    request.session['token'] = str(token)
+    request.session.set_expiry(1800) #session expires in 1800 seconds = 0.5 hour
```

Comment:
increase maybe? most apps ive used seem to preserve sessions for longer than half hour (except for financial ones for obvious reasons)

Reply:
Increased to 1 hr.